common/src/main/java/com/microsoft/identity/common/internal/ui/embeddedwebview/EmbeddedWebViewAuthorizationStrategy.java

```diff
@@ -0,0 +1,126 @@
+import com.microsoft.identity.common.internal.providers.microsoft.microsoftsts.MicrosoftStsAuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationResult;
+import com.microsoft.identity.common.internal.providers.oauth2.AuthorizationStrategy;
+import com.microsoft.identity.common.internal.ui.AuthorizationConfiguration;
+
+
+public class EmbeddedWebViewAuthorizationStrategy extends AuthorizationStrategy {
+    private WebView mWebView;
+    private String mStartUrl;
+
+    public AuthorizationResult requestAuthorization (AuthorizationRequest request) {
+        throw new UnsupportedOperationException("Not implemented yet.");
+    }
+
+    EmbeddedWebViewAuthorizationStrategy(Activity activity, AuthorizationRequest request) {
+        if (activity == null || request == null) {
+            throw new IllegalArgumentException("Null activity or request");
+        }
+
+        //TODO validate auth request
+        setupWebView(activity, request);
+        setupStartURL();
+
+    }
+
+    private void setupWebView(final Activity activity, final AuthorizationRequest request) {
+        // Create the Web View to show the page
+        mWebView = (WebView)activity.findViewById(activity.getResources().getIdentifier("webView1", "id",
+                activity.getPackageName()));
+        mStartUrl = "about:blank";
+        mWebView.getSettings().setUserAgentString(
+                mWebView.getSettings().getUserAgentString() + AuthenticationConstants.Broker.CLIENT_TLS_NOT_SUPPORTED);
+
+        mWebView.getSettings().setJavaScriptEnabled(true);
+        mWebView.requestFocus(View.FOCUS_DOWN);
+
+        // Set focus to the view for touch event
+        mWebView.setOnTouchListener(new View.OnTouchListener() {
+            @Override
+            public boolean onTouch(View view, MotionEvent event) {
+                int action = event.getAction();
+                if ((action == MotionEvent.ACTION_DOWN || action == MotionEvent.ACTION_UP) && !view.hasFocus()) {
+                        view.requestFocus();
+                }
+                return false;
+            }
+        });
+
+        mWebView.getSettings().setLoadWithOverviewMode(true);
+        mWebView.getSettings().setDomStorageEnabled(true);
+        mWebView.getSettings().setUseWideViewPort(true);
+        mWebView.getSettings().setBuiltInZoomControls(true);
+        mWebView.setVisibility(View.INVISIBLE);
+
+        selectWebViewClient(activity, request);
+    }
+
+    private void selectWebViewClient (final Activity activity, final AuthorizationRequest request) {
+        // if it is an ADAL request, we set the AADWebViewClient
+        // if it is a MSAL request, we set the MSSTSWebviewClient
+        // if it is a broker request, we set the BrokerWebviewClient and broker always talk to MSAL
+
+        if(AuthorizationConfiguration.getInstance().isADALRequest() &&

```

Comment:
This feels like a problem to solve polymorphically through delegation to another object provided by ADAL or MSAL via interface, rather than by checking types. Common 'knows too much'

Reply:
Removing this function. As the webviewclient does not need to distinguish v1 from v2. The logic is handled in AuthroizationRequest and AuthroizationResult layer.